tests/phpunit/tests/test-crud-posts.php

```diff
@@ -245,4 +245,34 @@ public function test_existing_post_updated_with_tag_mixing_slug_and_name() {
 		$this->assertCount( 1, $tags );
 		$this->assertSame( 'Common', reset( $tags )->name );
 	}
+
+	/**
+	 * Test a post creation with a non-existent default category to check that no category has been assigned to this post.
+	 *
+	 * @ticket 2248.
+	 * @see https://github.com/polylang/polylang-pro/issues/2249.
+	 */
+	public function test_save_post_when_no_default_category_set() {
+		$this->pll_admin->posts = new PLL_CRUD_Posts( $this->pll_admin );
+
+		// Create an english category.
+		$en_cat = self::factory()->term->create( array( 'taxonomy' => 'category', 'name' => 'English category' ) );
+		$this->pll_admin->model->term->set_language( $en_cat, 'en' );
+
+		// Create a post with this category so that `get_terms()` returns terms (since `hide_empty` is `true` by default).
+		self::factory()->post->create( array( 'post_category' => array( $en_cat ) ) );
+
+		update_option( 'default_category', '999' );
```

Comment:
Some day, someone unlucky will have `$en_cat === 999`. FYI I just got `1010` in my first test (so very close). So you must make sure that this will never happen. Either use a really huge number or something like `$en_cat + 999` that cannot exist yet. Also add a comment to explain why this number.